embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -144,9 +152,46 @@ protected ClientConfiguration getClientConfiguration(PluginTask task)
         clientConfig.setMaxErrorRetry(3); // SDK default: 3
         clientConfig.setSocketTimeout(8*60*1000); // SDK default: 50*1000
 
+        // set http proxy
+        if (task.getHttpProxy().isPresent()) {
+            setHttpProxy(clientConfig, task.getHttpProxy().get());
+        }
+
         return clientConfig;
     }
 
+    private Optional<HttpProxy> setupHttpProxy(PluginTask task)
```

Comment:
`setupHttpProxy` sounds like the configuration is actually set by this method, but it's actually just constructing a `HttpProxy` value with a default value if not existing.

How about `createHttpProxyWithDefault` or `construct...`?